### DISCUSSION TITLE
Files are saved as folders containing xl.meta files for minio docker container

### DISCUSSION THREAD

**Discussion by TheSharpOwl** [minio/minio#17574] [Q&A] [answered]
### Describe the bug

When uploading a file (image png/jpg or json file) using a string stream, let's say `sample.json` I get a folder called `sample.json` and inside of it there is `xl.meta` file.

I tried to save the image locally and it was working totally okay so the encoder code is working. Also, I remember one year ago it was saving on minio container fine but now I came back to the same exact code one year ago but still the issue arises.

Also it was discussed in github on AWS C++ sdk repo [here](https://github.com/aws/aws-sdk-cpp/issues/2561)

Conditions:
* Docker image for minio to connect to

Where it was tried:
* Ubuntu 18
* Windows 10

How it can be seen
* Using a terminal and trying to `cd` the image name
* Using volumes mounting

### Expected Behavior

Get an image/json file not a folder

### Current Behavior

I get a folder called `sample.png` and inside of it there is `xl.meta` file.

### Reproduction Steps

docker image command line:
```bash
docker run -p 9000:9000 --name minio  -e "MINIO_ACCESS_KEY=minioadmin" -e "MINIO_SECRET_KEY=miniokey" -v C:/aws/data:/data  minio/minio server /data
```

```cpp
// connect to a bucket using the examples then using this method:
bool saveData(const char *rawData,
                           size_t dataSize,
                           const std::string &filename) {
  Aws::S3::Model::PutObjectRequest putObjectRequest;
  putObjectRequest.WithBucket(bucketName_.c_str()).WithKey(filename.c_str());

  auto data = Aws::MakeShared<Aws::StringStream>(
      "PutObjectInputStream",
      std::stringstream::in | std::stringstream::out
          | std::stringstream::binary);

  data->write(rawData, dataSize);

  putObjectRequest.SetBody(data);
   // these lines were tried too and they didn't fix it
  // putObjectRequest.SetContentType("image/jpeg");
  // putObjectRequest.SetContentLength(dataSize);
  auto result = client_->PutObject(putObjectRequest);
  if (result.IsSuccess()) {
    return true;
  }
  return false;
}
```

### AWS CPP SDK version used

1.11.65 (vcpkg)

### Compiler and Version used

Visual Studio 2017 15.9.48 Community

### Operating System and version

Windows 10 19044.3086

**Comment by klauspost** [comment, marked as answer]
This is expected behavior. You cannot and should not access files outside the S3 API.

**Reply by TheSharpOwl** [reply]
But when I don't mount and go into docker terminal the same happens I can `cd` to the folder and see the meta file inside of it meaning that I don't actually get a text file or an image

**Reply by klauspost** [reply]
Yes - you cannot access objects through the file system as stated above.

**Reply by TheSharpOwl** [reply]
so you mean through docker bash I cannot access ? well I also tried to copy with `docker cp` from docker to my local pc and the same issue is happening.

Anyway, if it cannot be accessed using a file system and cannot copy it to my file system then what is the purpose of having files there? I cannot access them nor I can check if uploading is working :/

**Reply by klauspost** [reply]
MinIO provides an S3 API which allows full control of your objects. Use that for all operations.

There are many clients, including our own: [mc](https://min.io/docs/minio/linux/reference/minio-mc.html?ref=docs).

**Reply by TheSharpOwl** [reply]
Thank you very much it works now for me now )

**Reply by umerasghar-ibm** [reply]
Hi @TheSharpOwl , how you were able to fix this issue? can you guide pleas.e

**Reply by TheSharpOwl** [reply]
@umerasghar-ibm You should use [this](https://min.io/docs/minio/linux/reference/minio-mc.html?ref=docs) to copy the files (using this executable commands, you can copy the files or check them)

**Reply by lakuhana** [reply]
> MinIO provides an S3 API which allows full control of your objects. Use that for all operations.
> There are many clients, including our own: [mc](https://min.io/docs/minio/linux/reference/minio-mc.html?ref=docs).

I would like the developer to provide a configuration that lets me choose between using 'xl.meta' or my own file

**Comment by kisst** [comment]
While I understand the fact that the underlying storage of objects is not designed for direct access I see a valid use case for it. I can imagine a simple, lightweight Fuse based solution for at least read only mounts, even with the tradeoff of having no event triggers nor permission management.

**Reply by harshavardhana** [reply]
Sure feel free to work on it, let us know if you need any guidance.

**Comment by lakuhana** [comment]
> 有很多客户，包括我们自己的客户：[mc](https://min.io/docs/minio/linux/reference/minio-mc.html?ref=docs)。
I would like the developer to provide a configuration that lets me choose between using 'xl.meta' or my own file

**Reply by klauspost** [reply]
That is not going to happen.
